kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js

```diff
@@ -50,17 +53,44 @@ define([
             bus = runtime.bus().connect(),
             channel = bus.channel(config.channelName),
             ui,
-            dataSource = 'ftp_staging', // only option now, should come from the parameterSpec.
+            dd_options = spec.original.dynamic_dropdown_options,
+            dataSource = dd_options.data_source || 'ftp_staging',
             model = {
                 value: undefined
             },
             stagingService = new StagingServiceClient({
                 root: runtime.config('services.staging_api_url.url'),
                 token: runtime.authToken()
             }),
+            searchClient = new KBaseSearchEngine(
+                runtime.config('services.KBaseSearchEngine.url'),
+                {token: runtime.authToken()}
+            ),
             userId = runtime.userId(),
             eventListeners = [];
 
+        /**
+         * This function takes a nested return and returns a flat key-value pairing for use with
+         * handlebar replacement for example {"foo":{"bar": "meh"}} becomes {"foo.bar": "meh"}
+         */
+        var flattenObject = function(ob) {
+            var toReturn = {};
+            for (var i in ob) {
```

Comment:
Is `ob` always expected to be an object? If so, the `Object.keys().forEach()` idiom can be used here, too.